### PR TITLE
Migrate to gitlab-ci

### PR DESCRIPTION
**Pull request by SusVlaAlex**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,9 +1,70 @@
...
+variables:
+  VAULT_ADDR: "http://vault.sit-ota.aws.in.here.com"
+
+test:
+  stage: test
+  tags:
+    - docker-sock
+  image: advancedtelematic/gitlab-jobs:0.2.3
+  script:
+    - sbt -sbt-dir ./.sbt -ivy ./.ivy2 -J-Xmx1G -J-XX:ReservedCodeCacheSize=128m ota-plus-web/ut:test
```
Comment (simao) [inline .gitlab-ci.yml]: Lets use `sbt -sbt-dir ./.sbt -ivy ./.ivy2 ota-plus-web/ut:test`, if it works. In general we should not limit the memory anymore, I think this should work.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1,9 +1,70 @@
...
+  only:
+    - master
+  tags:
+    - docker-sock
+  image: advancedtelematic/gitlab-jobs:0.2.3
+  script:
+    - env | sort
+    - export VAULT_TOKEN=$(cat /opt/vault/token)
+    - gitlab-docker-login $VAULT_ADDR
+    - ./sbt -x -sbt-dir ./.sbt -ivy ./.ivy2 -batch ";project ota-plus-web;release" -J-Xmx1G -J-XX:ReservedCodeCacheSize=128m -Dsbt.log.noformat=true
```
Comment (simao) [inline .gitlab-ci.yml]: same

**Comment by houcros** [comment]
I think this should be part of the pipeline somehow: https://github.com/advancedtelematic/ota-plus-server/blob/162f72b612ac50be37038ead5cd9095c4aca7133/ota-plus-web/build.sbt#L30 to build the FE side. I remember building the docker image without that step resulted in a broken app.

**Comment by zabbal** [comment]
Btw, it might make sense to elaborate in the commit message: where do we migrate from?

File: ota-plus-web/build.sbt [hunk 3]
```diff
@@ -30,11 +30,7 @@ val runWebpack = taskKey[Seq[Int]]("Run webpack:10")
 runWebpack := {
   Seq(
     Process(
-      "docker" :: "run" ::
-        "--rm" ::
-        "--volume" :: s"${baseDirectory.value.toString}/app:/app" ::
-        "advancedtelematic/webpack:10.1" ::
-        "sh" :: "-c" :: "cd reactapp && npm install && webpack -p && cd ../reactapp-new && npm install && npm run build" ::
+    "sh" :: "-c" :: s"cd ${baseDirectory.value.toString}/app/reactapp && npm install && webpack -p" ::
```
[the remark this answers is not in the document]
Comment (houcros) [reply ota-plus-web/build.sbt]: Will this still successfully build the FE locally? Only if the devs have npm and webpack it seems? I guess we can live with that?

[the remark this answers is not in the document]
Comment (simao) [reply ota-plus-web/build.sbt]: I mean.. if you are a FE dev....

They are probably not even using this anyway

**Pull request merged**